pkg/artifactory/user.go

```diff
@@ -0,0 +1,211 @@
+package artifactory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type User struct {
+	Name                     string   `json:"name"`
+	Email                    string   `json:"email"`
+	Password                 string `json:"password,omitempty"`
+	Admin                    bool     `json:"admin"`
+	ProfileUpdatable         bool     `json:"profileUpdatable"`
+	DisableUIAccess          bool     `json:"disableUIAccess"`
+	InternalPasswordDisabled bool     `json:"internalPasswordDisabled"`
+	LastLoggedIn             string   `json:"lastLoggedIn"`
+	Realm                    string   `json:"realm"`
+	Groups                   []string `json:"groups"`
+}
+
+var baseUserSchema = map[string]*schema.Schema{
+	"name": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.StringIsNotEmpty,
+		Description:  "Username for user.",
+	},
+	"email": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ValidateFunc: validateIsEmail,
+		Description:  "Email for user.",
+	},
+	"admin": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "When enabled, this user is an administrator with all the ensuing privileges.",
```

Comment:
i guess, we started to specify Optional/Required in the description text. And, have the default value mentioned for optional fields. Can we do the same in this schema too?

Reply:
I haven't included 'Optional' or 'Required' in the description because the auto doc gen plugin that TF suggests (or one we may create ourselves) will introspect the code and append the text.

In the meantime, I can add them.